src/type/preload.ts

```diff
@@ -280,13 +281,40 @@ export type DefaultStyleId = {
   defaultStyleId: number;
 };
 
-export type MinimumEngineManifest = {
+export type supportedFeaturesItem = {
+  type: string;
+  value: boolean;
+  name: string;
+};
+
+export const supportedFeaturesItem = z.object({
+  type: z.string(),
+  value: z.boolean(),
+  name: z.string(),
+});
+
+export type SupportedFeatures = z.infer<typeof supportedFeaturesItem>;
+
+export type minimumEngineManifest = {
   name: string;
   uuid: string;
   command: string;
-  port: string;
+  port: number;
+  supported_features: Record<string, supportedFeaturesItem>;
 };
 
+export const minimumEngineManifest = z
+  .object({
+    name: z.string(),
+    uuid: z.string(),
+    command: z.string(),
+    port: z.number(),
+    supported_features: z.record(z.string(), supportedFeaturesItem),
```

Comment:
`snake_case`が出てくるのちょっと悲しいですね。。
本当はこうしたい、という意味を込めてFIXMEコメントを残してみるのはどうでしょう。
（結構調べたけど良い解決策はなさそうでした 😇 ）

```suggestion
    supported_features: z.record(z.string(), supportedFeaturesItem), // FIXME: camelCaseにする
```